.github/workflows/test-environment.yml

```diff
@@ -152,6 +152,18 @@ jobs:
           echo "::add-mask::$ec_api_key"
           echo "TF_VAR_ec_api_key=$ec_api_key" >> $GITHUB_ENV
 
+      - name: Process BC version
```

Comment:
I would consider getting the hash on a different optional parameter, `PIN_VERSION` for example.
This way we will avoid the cleaning code implemented both in bash and terraform, and the flow of using the pinned version will be more straight forward `PIN_VERSION != ""`.

Reply:
@amirbenun I have refactored the code to support a separate `pin_version` Terraform variable. This eliminates the need for cleanup version in Terraform code. I've updated the step action responsible for initializing and updating `TF_vars` and provided comments for clarity. Additionally, I have improved the documentation to make it more clear.